### ISSUE TITLE
Accessibility: Review and increase aria-live usage

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#11150]
## Task Description

We have this useful `useLiveRegion` hook but we don't make too much use of it unfortunately.

There are certainly quite a few places where we should be using it to announce changes (such as when adding or removing items from somewhere) to assistive technologies when they are not already otherwise announced (e.g. using a dialog or snackbar message).

For example:

* Dashboard: Error messages displayed for incorrect inputs on the settings page
* Editor: Error messages for page attachments conflicting with links
* Editor: Error messages for hotlinking (as pointed out in #11122)
* Editor: Error messages for link inputs
